### DISCUSSION TITLE
maximum core clock frequency (targeting icebreaker)

### DISCUSSION THREAD

**Discussion by biosbob** [stnolting/neorv32#587] [Q&A] [answered]
i was easily able to increase `CLOCK_FREQUENCY` to 36MHz in my top-level module (adjusting the `SB_PLL40_CORE` generics appropriately)....  everything worked fine in my test programs....

but when i tried 48MHz, things appeared to break....  note that i have a custom bootloader that copies from QSPI into IMEM; and while i'm dividing down my SPI CLK, perhaps i'm not able to access flash at this speed????

given my icebreaker board (12MHz source clock, QSPI), what's the maximum `CLOCK_FREQUENCY` i can achieve????

**Comment by GideonZ** [comment]
SPI clocks of 48 MHz are usually not a problem...

Did you set your timing constraints properly?
Did you set your FPGA speed grade correctly in the tool?
Do you have unconstrained paths?
Was the place and route tool able to achieve timing?

The maximum clock frequency that you can achieve depends on:
- Target architecture of the FPGA
- FPGA speed grade
- Synthesis options
- Degrees of freedom that PAR has (the fuller the FPGA has, the longer
routing paths will be)

**Reply by biosbob** [reply, marked as answer]
my bad....  the output from the synthesis contained the following line:

> Info: Max frequency for clock 'pll_clk': 24.65 MHz (FAIL at 36.00 MHz)

needless to say, it also failed when attempting 48.00 MHz!!!!!

(why i was able to run at 36 MHz may simply be a "red herring")

all good!!!!

**Reply by stnolting** [reply]
Great answer @GideonZ!

**Comment by agamez** [comment]
I can confirm that I get neorv32 on the icebreaker running at about 25MHz. Depending on the overall design I may have been lucky a couple of times and reached 26MHz of max frequency, but not more.

**Reply by stnolting** [reply]
I have a small NEORV32 config also running on an icebreaker board that operates at 27 MHz (using Lattice Radiant, Synplify Pro synthesis engine).
